sapi/cli/php_cli.c

```diff
@@ -1208,6 +1212,11 @@ int main(int argc, char *argv[])
 	 */
 	argv = save_ps_args(argc, argv);
 
+#ifdef PHP_WIN32
```

Comment:
Perhaps (why would you do that?), but I was basing this on the fact that the new codepage stuff uses this guard. Also, `_get_osfhandle` returns -2 if the handle is invalid (rather than -1).